### PR TITLE
filter tasks

### PR DESCRIPTION
**Pull request by mim24sd**
In this pull request I want to :
1) When You click on each tab in `nav` your tasks should be filtered.
2) State of selected tab should be stored in `localStorage`.

### CODE REVIEW

File: assests/javascript/index.js [hunk 1]
```diff
@@ -275,7 +294,20 @@ function formatDate(time) {
...
+  const state = localStorage.getItem("state");
+
+  if (state === "completedTasks") {
+    return getCompletedTasks(tasks);
+  } else if (state === "unCompletedTasks") {
+    return getUncompletedTasks(tasks);
+  }
+  else {
+    return tasks;
+  }
```
Comment (blacksrc) [inline assests/javascript/index.js]: code format!

File: assests/javascript/index.js [hunk 2]
```diff
@@ -330,3 +362,245 @@ function hideInputError() {
...
+
+// function showInputError(errorText) {
+//   addTaskInputErrorMassage.innerText = errorText;
+//   addTaskBox.classList.add("error");
+// }
+
+// function hideInputError() {
+//   addTaskInputErrorMassage.innerText = "";
+//   addTaskBox.classList.remove("error");
+// }
```
Comment (blacksrc) [inline assests/javascript/index.js]: Is this necessary?

File: assests/javascript/index.js [hunk 3]
```diff
@@ -6,10 +6,13 @@ const addTaskBox = document.getElementById("add-task-box");
 const addTaskInputErrorMassage = document.getElementById("title-error");
 const networkErrorMassage = document.getElementById("network-error");
 const taskList = document.getElementById("task-list");
+const headers = { "Content-Type": "application/json" };
```
Comment (blacksrc) [inline assests/javascript/index.js]: move this to the top or bottom of the imports and put a new line in between.

File: assests/javascript/index.js [hunk 4]
```diff
@@ -23,7 +26,22 @@ addTaskIcon.addEventListener("click", () => {
...
+
+completedTasks.addEventListener("click", () => {
+  localStorage.setItem("state", "completedTasks");
+  showTasks();
+});
+
+unCompletedTasks.addEventListener("click", () => {
+  localStorage.setItem("state", "unCompletedTasks");
+  showTasks();
+});
```
Comment (blacksrc) [inline assests/javascript/index.js]: Nice 🔥

File: assests/javascript/index.js [hunk 5]
```diff
@@ -23,7 +26,22 @@ addTaskIcon.addEventListener("click", () => {
   clearNewTaskInput();
 });
 
-showAllTasks();
+allTasks.addEventListener("click", () => {
+  localStorage.setItem("state", "allTasks");
+  showTasks();
+});
```
Comment (blacksrc) [inline assests/javascript/index.js]: 1. Rename `state` to something meaningful like `task-list-filter`.
2. There is no if condition for this `allTasks` in `classifyTasks`. I suggest to just remove `task-list-filter` in this case.

File: assests/javascript/index.js (hunk 2 above)
Comment (mim24sd) [reply assests/javascript/index.js]: Oh Sorry I totally forgot to remove this after adding filter parts of that.
Thanks

File: assests/javascript/index.js (hunk 3 above)
Comment (mim24sd) [reply assests/javascript/index.js]: ok

File: assests/javascript/index.js (hunk 4 above)
Comment (mim24sd) [reply assests/javascript/index.js]: thankssss

File: assests/javascript/index.js (hunk 5 above)
Comment (mim24sd) [reply assests/javascript/index.js]: I did first but didn't get the second one

**Review by blacksrc — APPROVED** [review]
✅

**Pull request merged**